graylog2-server/src/main/java/org/graylog/plugins/map/geoip/GeoIpResolverEngine.java

```diff
@@ -18,78 +18,120 @@
 
 import com.codahale.metrics.MetricRegistry;
 import com.codahale.metrics.Timer;
-import com.google.auto.value.AutoValue;
 import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableMap;
 import com.google.common.net.InetAddresses;
-import com.maxmind.geoip2.DatabaseReader;
-import com.maxmind.geoip2.model.CityResponse;
-import com.maxmind.geoip2.record.City;
-import com.maxmind.geoip2.record.Country;
-import com.maxmind.geoip2.record.Location;
+import org.apache.commons.lang3.StringUtils;
 import org.graylog.plugins.map.config.GeoIpResolverConfig;
 import org.graylog2.plugin.Message;
+import org.graylog2.utilities.ReservedIpChecker;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import javax.annotation.Nullable;
-import java.io.File;
-import java.io.IOException;
 import java.net.InetAddress;
-import java.nio.file.Files;
+import java.util.List;
+import java.util.Locale;
 import java.util.Map;
-import java.util.Optional;
+import java.util.stream.Collectors;
 
 import static com.codahale.metrics.MetricRegistry.name;
 
 public class GeoIpResolverEngine {
     private static final Logger LOG = LoggerFactory.getLogger(GeoIpResolverEngine.class);
 
-    private final Timer resolveTime;
-    private DatabaseReader databaseReader;
+    /**
+     * A mapping of fields that to search that contain IP addresses.  ONLY these fields will be checked
+     * to see if they have valid Geo IP information.
+     *
+     * <p>
+     * The mapping is <b>field name</b> -> <b>new message field prefix</b>, where field_name is the field name expected in the message
+     * which will be searched in the GeoIP database, and the new message field prefix is the prefix for the new field with the GeoIP data
+     * that will be inserted into the message.
+     * </p>
+     */
```

Comment:
Not yet.  Some of these finer points are still up in the air until we can discuss and determine how to best proceed.

Reply:
I discussed this with Product Management and the proposed solution was to add a configuration option on whether to use the official schema fields only or to use any field--by default it will look at all fields as before.